### ISSUE TITLE
Obfuscation of votes during beta period to protect beta users

### ISSUE THREAD

**Issue by Voileexperiments** [Codewars/codewars.com#3052]
In many years of doing beta process in CW I have encountered no less than several cases of hostile kata authors or dodgy power users/moderators de-anonymizing my ranking/satisfaction votes through various means, then proceed to announce and use this information to perform public shaming or/and personal attack on my voting. (This has not just happened to myself; victims include ZED.CWT and others.)

This is unacceptable, borderline security/privacy issue and destroys the integrity of the beta voting process. Having most users who does beta process in the same Discord that interacts every day and has close acquaintance to each other is already very questionable and drama-prone; yet there are many instances of users who are willing to manipulate votes through various means to get kata satisfaction rate to approval percentage (common tactics from kata authors include sock puppeting, vote begging, emotional accusation, and, as mentioned above, bullying other beta users for not voting "Very Satisfied").

These actions make beta process useless as a means to curb katas until they can be approved: this probably explains partly why there are a lot of katas in approval with nobody dare to approve them, because it is implicitly understood that satisfaction rate is *not* reliable as a metric of a kata that wouldn't cause many problems after approval.

Therefore I'd like to propose obfuscating ranking/satisfaction votes during beta period, to protect the anonymity of users participating in beta from any potential drama caused by hostile users:

- During beta, the first solves are most prone to reverse engineering of votes. Therefore the rank and satisfaction votes should be completely obscured, besides the total amount of votes. 4 is probably enough (since 4 is also the threshold for beta kata retirement).
- After that, there are still slight possibilities to reverse engineer satisfaction votes. It should be mitigated by rounding down the satisfaction rate to the nearest 5/10% (so 80% kata approval threshold is still exact to the rounded value).
- After katas are approved, votes are much less important anyway, so obfuscation is mostly unnecessary.

**Comment by Voileexperiments** [comment]
Okay, so there is an example in the wild, **3 years ago**, where a power user is bullied by a kata author for making a ranking vote that is in line with the voter (aka it's genuine), but which the kata author disagrees with:

https://www.codewars.com/kata/5f24315eff32c4002efcfc6a/discuss#5f2535833ec3ea00297905f7

This is beyond ridiculous. Vote witch-hunting should NOT be allowed on CW in the first place; otherwise nobody are going to vote genuinely (in fear of being witch-hunted), and the integrity of the beta process is ruined. (Beta process is already kind of a joke; this would make it even worse)

This kata author clearly un-anonymized the vote by seeing who made the first solve, and what the ranking/satisfaction vote ended up as a result. This means that the current vote system is not robust enough to defend beta users against such hostile kata authors.

**Comment by Blind4Basics** [comment]
Don't forget to state the numerous cases where some users are just shit-ranking, also. Like...

> victims include ZED.CWT

You're sort of instantly discarding your point with this sentence alone, because the _beta process_ has been victim of ZED for years, not the other way around.

The situation is not ideal, for sure, but your base assumption looks like voters actually know what they are doing and rank honestly. But that's not always true. In some cases, it's even outright wrong.

Also, what do we do with troll users? They are _also_ bullying the beta process.

**Comment by Blind4Basics** [comment]
oh, btw... Those two "bullied users", like you say, are two notorious _trolls_ when it comes to ranking. There is no other word. Again, you're discarding your entire point with that example.

**Comment by Voileexperiments** [comment]
> Don't forget to state the numerous cases where some users are just shit-ranking

Who is to determine which user is "shit-ranking"? According to your opinions?

The vote is supposed to be a democratic process; nobody gets to say whose vote is legitimate and who is not, especially not you; *the site* gets to decide that. "ZWT's votes are trolling" are purely your opinion and AFAIK a lot of people disagrees with you on this.

> You're sort of instantly discarding your point with this sentence alone, because the beta process has been victim of ZED for years, not the other way around.

This is the real shit-flinging and witch-hunting I'm referring to here, and you're exactly doing that.

Why are you wasting your time insisting on your opinions on what votes are right and preaching to the choir, when the whole voting process should boil down to "if you don't like it, fucking vote the opposite"?  
The majority decides the vote. That's how the voting system works. If there is a 8kyu vote, what's the matter? Once it reaches beta threshold there will be several 4kyu votes casted anyway, so the whole thing balances out.

In the grand scheme of "voting", only the process matters and nobody cares about what you think about the other voters and how they voted. Why are you so obsessive towards how individuals voted and "proving" how others are voting "ingeniously"? You've literally become a threat to the voting system with this act.

> The situation is not ideal, for sure, but your base assumption looks like voters actually know what they are doing and rank honestly. But that's not always true. In some cases, it's even outright wrong.

Thanks, captain obvious! Now tell me how you're going to determine if someone is "ranking honestly"? According to your own judgement?  
You're just making up this stuff, judging others and then insist your judgement is objective and right.  
This is a deadlock that will never resolve: you'll never agree to ZED or my votes, ZED will never agree to your or my votes, and I'll never agree to his or your votes.

> Also, what do we do with troll users? They are also bullying the beta process.

I believe determining the actual trolls users (aka those provably voted maliciously) is much easier than trying to interrogate a user why they voted in a specific way, and then proceed to shove words into their mouths.  
(Assuming CW is competent anyway, lots of cheaters submitted cheating solutions to beta katas without submitting final, inflating the completion count, and there are no ways to track them. *Those* are the undisputably "troll users", and their stats are still around.)

Or, are you trying to say "ZWT's votes are 'troll votes' because ZWT is a 'troll user' because I say so"? If that's true I have problems with you as an official mod of CW. This is not acceptable behaviour for someone representing CW as a mod.

---

In short, as usual, if you aren't able to muster anything more than "I rely purely on my opinion to oppose on Voile on this topic", please do better. You're wasting both your and our time, and you haven't really made any compelling points. *And* this is not the first time this has happened. Please bring in better points than what you brought last time we argued about this.

**Comment by Blind4Basics** [comment]
x)

> "ZWT's votes are trolling" are purely your opinion and AFAIK a lot of people disagrees with you on this.

that's your opinion, and a lot of people also disagree with you...

When someone puts a white rank vote on a kata that ends up in blue or purple level, there is no "personal opinion" involved. Same when a user rank yellow a kata which ends up approved in purple level.

**Comment by Voileexperiments** [comment]
> When someone puts a white rank vote on a kata that ends up in blue or purple level, there is no "personal opinion" involved. Same when a user rank yellow a kata which ends up approved in purple level.

I hope you have heard of [Sliding Puzzle Solver](https://www.codewars.com/kata/5a20eeccee1aae3cbc000090), the easiest 1kyu ever. Yeah that's really a 5kyu kata...

**Comment by Blind4Basics** [comment]
yeah, just slightly easier than immortal.

also, for your information: you're saying that myjinxin and kontic do not know what they do when they rank in the beta process? (the third voter, who is the one who voted 2kyu, is unknown to me)

**Comment by Voileexperiments** [comment]
> you're saying that myjinxin and kontic do not know what they do when they rank in the beta process?

First, I never specifically pointed out which individual voted what. I only pointed out that *I do not agree with the overall ranking of the kata*, and that if I were voting it I'd be voting 5kyu.

I don't know why you seem to like de-anaonymizing votes and make votes personal. Again, I should point out that in a democratic voting process, tracing every vote to specific individuals is a threat to the system, and nobody should be doing this, *specifically* because of this behaviour you've been doing.

Second, while I do agree both this and Immortal are very overrated, that's what the beta testers have voted.  
Following your logic of "whoever not voted what I think is right is wrong", you should yell at the beta testers for making "wrong votes"/"troll votes", then?  
These are not the first time beta voters are making absurdly overranked/underranked votes; if you choose to ignore all of those, and instead of proposing something infinite more sane (such as actually making an *absolute* reference of each kyu's level, so we don't have to follow the herd when beta-voting) you proceed to specifically come and slam down me/ZWT/certain other people with nothing but "because I say so" and "because it is true" to support yourself, you cannot expect me to not retaliate to your double standard.

I do not believe this behaviour is appropriate **for a person in the role of a CW mod**. You are a disgrace to CW's image.

---

Also, coming back to this:

> When someone puts a white rank vote on a kata that ends up in blue or purple level, there is no "personal opinion" involved. Same when a user rank yellow a kata which ends up approved in purple level.

So you are saying "if someone is voting deviating to the norm it is perfectly normal to disregard their opinion"? That's... a very blatant form of discrimination/racism.  
I hope you know what you're doing with this, because you're shooting yourself in the foot. This does not fly at all in a modern society.

**Comment by Blind4Basics** [comment]
Be serious...

You want me to be demoted? Don't hesitate to escalate your request. Don't forget to quote that part:

> You are a disgrace to CW's image.

In the end you don't have a problem with the votes, you have a problem with me. Nothing more. So just say it and don't use straw men as intermediates.

**Comment by VanessaGreene** [comment]
@Voileexperiments Apologies I tagged this and thought I had responded. I would like to discuss this further with you, can you please email me @ vanessa.greene@andela.com?
